src/events/requests/NewRequestEventHandler.ts

```diff
@@ -1,22 +1,77 @@
-import { Message } from 'discord.js';
+import { Message, TextChannel, RichEmbed } from 'discord.js';
 import * as log4js from 'log4js';
 import EventHandler from '../EventHandler';
+import BotConfig from '../../BotConfig';
+import { ReactionsUtil } from '../../util/ReactionsUtil';
+import MentionCommand from '../../commands/MentionCommand';
 
 export default class NewRequestEventHandler implements EventHandler {
 	public readonly eventName = '';
 
 	private logger = log4js.getLogger( 'NewRequestEventHandler' );
 
+	/**
+	 * A map from request channel IDs to internal channel objects.
+	 */
+	private readonly internalChannels: Map<string, TextChannel>;
+
+	constructor( internalChannels: Map<string, TextChannel> ) {
+		this.internalChannels = internalChannels;
+	}
+
 	// This syntax is used to ensure that `this` refers to the `NewRequestEventHandler` object
-	public onEvent = ( message: Message ): void => {
-		this.logger.info( `User ${ message.author.tag } posted a new request to requests channel ${ message.channel.id }` );
+	public onEvent = async ( origin: Message ): Promise<void> => {
+		if ( origin.type !== 'DEFAULT' ) {
+			if ( origin.deletable ) {
+				origin.delete();
+				return;
+			}
+		}
+
+		this.logger.info( `User ${ origin.author.tag } posted a new request to requests channel ${ origin.channel.id }` );
+
+		await origin.clearReactions();
 
-		if ( message.type === 'PINS_ADD' ) {
-			if ( message.deletable ) {
-				message.delete();
+		const regex = new RegExp( `(?:${MentionCommand.ticketLinkRegex.source}|${MentionCommand.ticketIdRegex.source})`, 'g' );
+
+		if ( BotConfig.request.no_link_emoji && !origin.content.match( regex ) ) {
+			origin.react( BotConfig.request.no_link_emoji );
+			const embed = new RichEmbed()
+				.setColor( '#F7C6C9' )
+				.setAuthor( origin.author.tag, origin.author.avatarURL )
+				.setDescription( `Your [request](${ origin.url }) doesn't not contain any valid ticket reference.` )
+				.setTimestamp( new Date() );
```

Comment:
Is it necessary to send an embed here? After all, the person posting the request should always immediately see it, so a ticket is not needed. Perhaps A ping could be included as well.

Maybe the bot could also mention that the user can also edit their message and does not need to delete it, along the lines of "If you want to add it, you can edit your message".

Also, is there any particular reason for using that precise color for the embed?

Reply:
> is there any particular reason for using that precise color for the embed
Nope. I just didn't know which color to use, so I basically picked it from my avatar